tensorpipe/common/tensor.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#if TENSORPIPE_HAS_CUDA_IPC_CHANNEL // TODO: Generic TENSORPIPE_HAS_CUDA.
+#include <cuda_runtime.h>
+#endif // TENSORPIPE_HAS_CUDA_IPC_CHANNEL
+
+namespace tensorpipe {
+
+enum class DeviceType {
+  kCpu = 0,
+#if TENSORPIPE_HAS_CUDA_IPC_CHANNEL
+  kCuda,
+#endif // TENSORPIPE_HAS_CUDA_IPC_CHANNEL
+};
+
+struct CpuTensor {
```

Comment:
I may be overthinking this, but should we assume that these structs are always copiable (and always will be, even for future device types)? Or should we consider that some may be movable-only and thus use std::move(...) whenever we pass them around? For now I don't really see any case in which moving would be required, so I guess we shouldn't worry about this right now...

Reply:
I'd be in favor of keeping them copyable, as they're intended as being lightweight. Should we change our stance, the compiler will tell us about every disallowed copy happening.

Reply:
Agreed.